### PR TITLE
[Tutorial] Add a tutorial for PyTorch

### PR DESCRIPTION
**Pull request by alexwong**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

Wasn't sure if a tutorial is expected for new frontends but I wrote one a while ago and now #4497 is merged.

### CODE REVIEW

File: tutorials/frontend/from_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+This article is an introductory tutorial to deploy PyTorch models with Relay.
+
+For us to begin with, PyTorch should be installed.
+TorchVision is also required since we will be using it as our model zoo.
+
+A quick solution is to install via pip
+
+.. code-block:: bash
+
+    pip install torch==1.4.0
```
Comment (kevinthesun) [inline tutorials/frontend/from_pytorch.py]: Maybe we can add the versions we support?

Comment (alexwong) [reply tutorials/frontend/from_pytorch.py]: Added a couple sentences below.

File: tutorials/frontend/from_pytorch.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+# numpy, packaging
+import numpy as np
+from packaging import version
+from tvm.contrib.download import download_testdata
+
+# PyTorch imports
+import torch
+import torchvision
+if (version.parse(torch.__version__) > version.parse("1.4.0")) \
```
Comment (yzhliu) [inline tutorials/frontend/from_pytorch.py]: should we move the check to the frontend?

Comment (alexwong) [reply tutorials/frontend/from_pytorch.py]: I would be okay with this but I didn't see this in any of the other frontends.

Comment (masahi) [reply tutorials/frontend/from_pytorch.py]: We don't want to update this script every time we bump the supported version, so it is better to remove it. I am using Torch source build with version 1.5, and I don't have any issue with our frontend.

File: tutorials/frontend/from_pytorch.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+# -------------------------------
+model_name = 'resnet18'
+model = getattr(torchvision.models, model_name)(pretrained=True)
+model = model.float().eval()
+
+# We grab the TorchScripted model via tracing
+input_shape = [1, 3, 224, 224]
+input_data = torch.randn(input_shape).float()
+scripted_model = torch.jit.trace(model, input_data).float().eval()
+
```
Comment (masahi) [inline tutorials/frontend/from_pytorch.py]: Remove .float() to reduce clutter

File: tutorials/frontend/from_pytorch.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+mod, params = relay.frontend.from_pytorch(scripted_model,
+                                          shape_dict)
+
+######################################################################
+# Relay Build
+# -----------
+# Compile the graph to llvm target with given input specification.
+target = 'llvm'
+target_host = 'llvm'
+layout = None
```
Comment (masahi) [inline tutorials/frontend/from_pytorch.py]: Remove layout

File: tutorials/frontend/from_pytorch.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+input_shape = [1, 3, 224, 224]
+input_data = torch.randn(input_shape).float()
+scripted_model = torch.jit.trace(model, input_data).float().eval()
+
+######################################################################
+# Load a test image
+# -----------------
+# Classic cat example!
+from PIL import Image
+img_url = 'https://raw.githubusercontent.com/Cadene/pretrained-models.pytorch/master/data/cat_224.jpg'
```
Comment (masahi) [inline tutorials/frontend/from_pytorch.py]: Better to use more reliable source. See for example https://github.com/apache/incubator-tvm/blob/0e189f01146c004f8a55840bb6c36d4449a25c34/tests/python/frontend/tflite/test_forward.py#L61

File: tutorials/frontend/from_pytorch.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                         std=[0.229, 0.224, 0.225])
+])
+img = my_preprocess(img)
+img = np.expand_dims(img, 0)
+
+######################################################################
+# Import the graph to Relay
+# -------------------------
+# Convert PyTorch graph to Relay graph.
+shape_dict = {'input0': img.shape}
```
Comment (masahi) [inline tutorials/frontend/from_pytorch.py]: The name 'input0' is arbitrary right? I think it is confusing and problematic if there are multiple inputs. Similarly to other frontends, we should use input names that come with input IR. I will fix this in my PR

File: tutorials/frontend/from_pytorch.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+class_id_to_key = [x.strip() for x in class_id_to_key]
+
+# Get top-1 result for TVM
+top1_tvm = np.argmax(tvm_output.asnumpy()[0])
+tvm_class_key = class_id_to_key[top1_tvm]
+
+# Convert input to PyTorch variable and get PyTorch result for comparison
+torch_img = torch.from_numpy(img)
+from torch.autograd import Variable
+torch_img = Variable(torch_img).float()
```
Comment (masahi) [inline tutorials/frontend/from_pytorch.py]: No need to wrap tensors with Variable, remove it

File: tutorials/frontend/from_pytorch.py [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+output = model(torch_img)
```
Comment (masahi) [inline tutorials/frontend/from_pytorch.py]: Add no_grad and remove detach below

File: tutorials/frontend/from_pytorch.py (hunk 5 above)
Comment (alexwong) [reply tutorials/frontend/from_pytorch.py]: I don't think TorchVision has any official source for class names and sample inputs. I'm using the class names provided in this repo and it also happened to have an input image as well. I'll take another look and find an alternative though.

File: tutorials/frontend/from_pytorch.py (hunk 6 above)
Comment (alexwong) [reply tutorials/frontend/from_pytorch.py]: Yes it is arbitrary. And yes, that is true and would probably simplify some of the parsing logic as well. I renamed it to img if that's slightly more clear.

File: tutorials/frontend/from_pytorch.py (hunk 5 above)
Comment (masahi) [reply tutorials/frontend/from_pytorch.py]: If there is no better option, adopt the same url used by other tutorial scripts

**Comment by masahi** [comment]
Thanks @alexwong @zhiics @comaniac @kevinthesun @yzhliu
